src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/begrunnelser/FinnGyldigeBegrunnelserForPeriodeContext.kt

```diff
@@ -0,0 +1,213 @@
+                        it.fom == vedtaksperiode.fom
+                    }?.verdi
+                if (person != null && forskøvedeVilkårResultaterMedSammeFom != null) {
+                    Pair(person, forskøvedeVilkårResultaterMedSammeFom)
+                } else {
+                    null
+                }
+            }.toMap().filterValues { it.isNotEmpty() }
+
+    private fun finnPersonerMedVilkårResultaterSomGjelderRettFørPeriode(): Map<Person, List<VilkårResultat>> =
+        personResultater.tilFørskjøvetVilkårResultatTidslinjeMap(persongrunnlag).mapNotNull { (aktør, tidsjlinje) ->
+            val person = persongrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
+            val forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode =
+                tidsjlinje.tilPerioderIkkeNull().singleOrNull {
+                    it.tom?.plusDays(1) == vedtaksperiode.fom
+                }?.verdi
+
+            if (person != null && forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode != null) {
+                Pair(person, forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode)
+            } else {
+                null
+            }
+        }.toMap().filterValues { it.isNotEmpty() }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPersonerUtenUtbetalingVedInnvilget(vedtakBegrunnelseType: VedtakBegrunnelseType) =
+        this.filterKeys {
+            if (vedtakBegrunnelseType == VedtakBegrunnelseType.INNVILGET) {
+                aktørIderMedUtbetaling.contains(it.aktør.aktørId) || it.type == PersonType.SØKER
+            } else {
+                true
+            }
+        }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPåVilkårType(vilkårTyperFraSanity: List<Vilkår>) =
+        this.mapValues { (_, vilkårResultaterForPerson) ->
+            vilkårResultaterForPerson.filter { vilkårTyperFraSanity.contains(it.vilkårType) }
+        }.filterValues { it.isNotEmpty() }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPåTriggere(
+        triggereFraSanity: List<Trigger>
+    ) = this.filter { (person, vilkårResultaterForPerson) ->
+        Trigger.values().filter { it.erOppfylt(vilkårResultaterForPerson, person) } == triggereFraSanity
+    }.filterValues { it.isNotEmpty() }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPåUtdypendeVilkårsvurdering(utdypendeVilkårFraSanity: List<UtdypendeVilkårsvurdering>) =
+        this.filterValues { vilkårResultaterForPerson ->
+            val utdypendeVilkårIBehandling =
+                vilkårResultaterForPerson.flatMap { it.utdypendeVilkårsvurderinger }.toSet()
+
+            if (utdypendeVilkårIBehandling.isEmpty()) {
+                utdypendeVilkårFraSanity.toSet() == utdypendeVilkårIBehandling
+            } else {
+                utdypendeVilkårIBehandling.all { utdypendeVilkårFraSanity.contains(it) }
+            }
```

Comment:
Dersom vi har krysset av på sommerferie i Sanity ønsker vi vel å få opp begrunnelsen så lenge vi har sommerferie huket av i ks-sak også, uavhengig om andre utdypende vilkår er krysset av i ks-sak. 

Altså si vi har
`utdypendeVilkårIBehandling = [SOMMERFERIE, ADOPSJON]`
`utdypendeVilkårFraSanity = [SOMMERFERIE]` 
Så skal vel det være ok?

Kan det stemme at riktig logikk da blir
```suggestion
                utdypendeVilkårFraSanity.all { utdypendeVilkårIBehandling.contains(it) }
```

Reply:
For alle begrunnelser som ikke er blandt de 4 standard-innvilgelsestekstene så stemmer det at vi kan bruke
```
utdypendeVilkårFraSanity.all { utdypendeVilkårIBehandling.contains(it) }
```
Men for de 4 så blir denne ☝️ litt for snill, da vi med følgende scenario vil få 2 og ikke 1 begrunnelse ut:
```
// Begrunnelse = Ikke barnehage
utdypendeVilkårIBehandling = [ADOPSJON]
utdypendeVilkårFraSanity = []
// Begrunnelsen kommer med

// Begrunnelse = Ikke barnehage og adopsjon
utdypendeVilkårIBehandling = [ADOPSJON]
utdypendeVilkårFraSanity = [ADOPSJON]
// Begrunnelsen kommer med
```
Mulig vi her kan slenge inn en sjekk på hvorvidt SanityBegrunnelseType = `STANDARD` 🤔